pkg/api/helpers/download.go

```diff
@@ -245,48 +235,107 @@ func (d *Downloader) downloadPresignedPart(ctx context.Context, physicalAddress
 		buf = buf[:partSize]
 	}
 
-	_, err = io.ReadFull(resp.Body, buf)
+	var err error
+	for attempt := 0; attempt <= d.BodyRetries; attempt++ {
+		if attempt > 0 {
```

Comment:
Do you think that all errors should be retryable? 
1. HTTP requests are already being retried.
2. There are some unrelated failures, like OS operations, that we should retry.